In [1]:
# Render our plots inline
%matplotlib inline

import pandas as pd
import matplotlib.pyplot as plt

plt.rcParams['figure.figsize'] = (15, 10)

# 1.1 Reading data from a csv file

We're going to be looking some cyclist data from Montréal. Here's the [original page](http://donnees.ville.montreal.qc.ca/dataset/velos-comptage) (in French), but it's already included in this repository. We're using the data from 2012.

This dataset is a list of how many people were on 7 different bike paths in Montreal, each day.

Load the dataset using `read_csv`. As you can see several parameters are available. Specify `encoding='latin1'`. 
the data is localter at `../data/bikes.csv`.


Select the first 3 rows using the same mechanism as numpy row selection

Date;Berri 1;Brébeuf (données non disponibles);Côte-Sainte-Catherine;Maisonneuve 1;Maisonneuve 2;du Parc;Pierre-Dupuy;Rachel1;St-Urbain (données non disponibles)
0                   01/01/2012;35;;0;38;51;26;10;16;                                                                                                               
1                   02/01/2012;83;;1;68;153;53;6;43;                                                                                                               
2                 03/01/2012;135;;2;104;248;89;3;58;

You'll notice that this it was not properly loaded : all columns collapsed into one. We will use options of  `read_csv` to fix that

* change the column separator to a `;` with the `sep` option
* Parse the dates in the 'Date' column using `parse_date`
* Tell it that our dates have the day first instead of the month first
* Set the index to be the 'Date' column with `index_col`

# 1.2 Selecting a column

You can select a column (or several) like a dictionnary juste put the name of the column between a brack : 
`df['column_name']` or `df[['col1', 'col2', etc]]`

# 1.3 Plotting a column

Plot the the column using  `.plot()` 

We can see that, unsurprisingly, not many people are biking in January, February, and March, 

We can also plot all the columns of dataframe using the same mechanism. 

However it is a bit cluttered as you can see. Plot it again using the `subplots=True` parameter

<style>
    @font-face {
        font-family: "Computer Modern";
        src: url('http://mirrors.ctan.org/fonts/cm-unicode/fonts/otf/cmunss.otf');
    }
    div.cell{
        width:800px;
        margin-left:16% !important;
        margin-right:auto;
    }
    h1 {
        font-family: Helvetica, serif;
    }
    h4{
        margin-top:12px;
        margin-bottom: 3px;
       }
    div.text_cell_render{
        font-family: Computer Modern, "Helvetica Neue", Arial, Helvetica, Geneva, sans-serif;
        line-height: 145%;
        font-size: 130%;
        width:800px;
        margin-left:auto;
        margin-right:auto;
    }
    .CodeMirror{
            font-family: "Source Code Pro", source-code-pro,Consolas, monospace;
    }
    .text_cell_render h5 {
        font-weight: 300;
        font-size: 22pt;
        color: #4057A1;
        font-style: italic;
        margin-bottom: .5em;
        margin-top: 0.5em;
        display: block;
    }
    
    .warning{
        color: rgb( 240, 20, 20 )
        }  